includes/reader-revenue/woocommerce/class-woocommerce-connection.php

```diff
@@ -160,20 +195,18 @@ public static function get_last_successful_order( $customer ) {
 			return false;
 		}
 
+		$user_id = $customer->get_id();
+
 		// Prioritize any currently active subscriptions.
-		$user_subscriptions = \wcs_get_users_subscriptions( $customer->get_id() );
-		if ( ! empty( $user_subscriptions ) ) {
-			foreach ( $user_subscriptions as $subscription ) {
-				if ( $subscription->has_status( self::ACTIVE_SUBSCRIPTION_STATUSES ) ) {
-					return $subscription;
-				}
-			}
+		$active_subscriptions = self::get_active_subscriptions_for_user( $user_id );
```

Comment:
This method returns an array of integers, which breaks the expected return of `get_last_successful_order()`. This causes the `get_contact_from_customer()` to throw a fatal error by trying to `->get_id()` from the integer and breaks data event handlers.

Reply:
Thanks for catching! I'll have a look and push an "alpha hotfix" for it

Reply:
PR here: https://github.com/Automattic/newspack-plugin/pull/3299